packages/hardhat-vyper/src/parser.ts

```diff
@@ -31,6 +33,16 @@ export class Parser {
     return result;
   }
 
+  private _validateTestModeNotUsed(fileContent: string, absolutePath: string) {
+    if (fileContent.includes('#@ if mode == "test":')) {
+      throw new Error(
```

Comment:
Let's throw a `VyperPluginError` instead. This error extends `NomicLabsHardhatPluginError`, which is then caught and handled by the Hardhat CLI.

Reply:
done